app/src/main/java/com/example/composetemplate/managers/FirebaseConfigurationManager.kt

```diff
@@ -0,0 +1,89 @@
+package com.example.composetemplate.managers
+
+import com.example.composetemplate.R
+import com.example.composetemplate.utils.Constants.Companion.CONFIG_FETCHED_FAILED
+import com.example.composetemplate.utils.Constants.Companion.CONFIG_FETCHED_SUCCESSFULLY
+import com.example.composetemplate.utils.Constants.Companion.CONFIG_UPDATED_FAILED
+import com.example.composetemplate.utils.Constants.Companion.CONFIG_UPDATED_SUCCESSFULLY
+import com.example.composetemplate.utils.LogsManager
+import com.example.composetemplate.utils.interfaces.Configurable
+import com.google.firebase.Firebase
+import com.google.firebase.remoteconfig.ConfigUpdate
+import com.google.firebase.remoteconfig.ConfigUpdateListener
+import com.google.firebase.remoteconfig.FirebaseRemoteConfig
+import com.google.firebase.remoteconfig.FirebaseRemoteConfigException
+import com.google.firebase.remoteconfig.remoteConfig
+import com.google.firebase.remoteconfig.remoteConfigSettings
+
+/**
+ * Manager class for handling Firebase Remote Config configurations.
+ *
+ * This class implements the [Configurable] interface, and is responsible for
+ * initializing, fetching, updating, and loading default configuration values
+ * from Firebase Remote Config. It ensures that configuration settings are
+ * fetched periodically and updates are handled dynamically.
+ * For using the data from the config we use FirebaseConfigProvider
+ */
+
+class FirebaseConfigurationManager : Configurable {
+
+    /* Firebase Remote Config instance used for fetching and managing configuration data. */
+    private val remoteConfig: FirebaseRemoteConfig = Firebase.remoteConfig
+
+    /* Configuration settings for Firebase Remote Config with a minimum fetch interval. */
+    private val configSettings = remoteConfigSettings {
+        /* Minimum interval for fetching new config data */
+        minimumFetchIntervalInSeconds = 3600
```

Comment:
maybe move it to constant? and explain its firebase recommendation so no one will change it

Reply:
I thought about it, but this value is only relevant here, and I don't want to use it anywhere else in the app.